returnn/datasets/meta.py

```diff
@@ -1557,11 +1555,17 @@ def _collect_single_seq(self, seq_idx):
                 if key in self.remove_in_between_postfix and sub_seq_idx != sub_seq_idxs[-1]:
                     assert data.ndim == 1 and data[-1] == self.remove_in_between_postfix[key]
                     data = data[:-1]
-                if key in self.repeat_in_between_last_frame_up_to_multiple_of and sub_seq_idx != sub_seq_idxs[-1]:
-                    multiple = self.repeat_in_between_last_frame_up_to_multiple_of[key]
-                    if data.shape[0] % multiple != 0:
-                        data = numpy.concatenate([data] + [data[-1:]] * (multiple - data.shape[0] % multiple), axis=0)
-                        assert data.shape[0] % multiple == 0
+                if key in self.force_align and sub_seq_idx != sub_seq_idxs[-1]:
+                    ref_key, frame_rate = self.force_align[key]
+                    ref_data = self.sub_dataset.get_data(sub_seq_idx, ref_key)
+                    len_diff = data.shape[0] - ref_data.shape[0] * frame_rate
+                    if len_diff > 0:
+                        # if data longer than ref_data * frame_rate, cut down
+                        data = data[:-len_diff,]
+                    elif len_diff < 0:
+                        # if data shorter than ref_data * frame_rate, pad by repeating last frame
+                        data = numpy.concatenate([data] + [data[-1:]] * -len_diff, axis=0)
```

Comment:
Should there be a sanity limit how much can be truncated or padded? Perhaps to allow padding/truncating only within one multiple of the target length vs. feature length? In my own experiments w/ subsampled MFCCs the mismatch was generally within this range (but then it was also MFCCs and not raw WAV...).

Reply:
Compared to features, for raw wav the mismatch will generally be larger by a factor of sampling_rate * frame_time e.g. 160 for 16kHz audio and 10ms frames.

Reply:
Do you mean sth like `assert len_diff <= frame_rate`? I think there is no clear upper bound if raw WAV is used. The mismatch is partially setup dependent(window_size - window_step in feature extraction) and can be more than one frame(10 ms). Maybe it would make sense to throw a warning instead of a hard limit?